src/common/Journald.cc

```diff
@@ -0,0 +1,280 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "Journald.h"
+
+#include <cstdio>
+#include <cstring>
+#include <endian.h>
+#include <fcntl.h>
+#include <memory>
+#include <sstream>
+#include <string>
+#include <sys/mman.h>
+#include <sys/socket.h>
+#include <sys/uio.h>
+#include <sys/un.h>
+#include <syslog.h>
+#include <unistd.h>
+
+#include "include/ceph_assert.h"
+#include "common/LogEntry.h"
+#include "log/Entry.h"
+#include "log/SubsystemMap.h"
+
+
+namespace ceph::logging {
+
+namespace {
+const struct sockaddr_un sockaddr = {
+  AF_UNIX,
+  "/run/systemd/journal/socket",
+};
+
+ssize_t sendmsg_fd(int transport_fd, int fd) {
+  constexpr size_t control_len = CMSG_LEN(sizeof(int));
+  char control[control_len];
+  struct msghdr mh = {
+    .msg_name = (struct sockaddr*)&sockaddr,
```

Comment:
please want to note, this is a GCC extension which allows us to use a C99 feature in C++. Designated initializers are not available in C++17 yet even though it is accepted in C++20.

Reply:
Thanks. Do you think is it ok to use designated initializers here? Since the newest gcc, clang, MSVC all support this.

Reply:
i think it's fine as long as you really believe it is necessary.

Reply:
```
/build/ceph-17.0.0-1747-gf8dd273d/src/common/Journald.cc: In function 'ssize_t ceph::logging::{anonymous}::sendmsg_fd(int, int)':
/build/ceph-17.0.0-1747-gf8dd273d/src/common/Journald.cc:42:3: sorry, unimplemented: non-trivial designated initializers not supported
   };
   ^
/build/ceph-17.0.0-1747-gf8dd273d/src/common/Journald.cc:42:3: sorry, unimplemented: non-trivial designated initializers not supported
src/common/CMakeFiles/common-common-objs.dir/build.make:377: recipe for target 'src/common/CMakeFiles/common-common-objs.dir/Journald.cc.o' failed
make[4]: *** [src/common/CMakeFiles/common-common-objs.dir/Journald.cc.o] Error 1
```

Reply:
OK, I will use traditional initialization then.

Reply:
OK, I will use traditional initialization then.